Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,171 @@
+                    case ExpressionType.LessThan:
+                        comparisonNode = new LtNode();
+                        break;
+                    case ExpressionType.LessThanOrEqual:
+                        comparisonNode = new LteNode();
+                        break;
+                    case ExpressionType.GreaterThan:
+                        comparisonNode = new GtNode();
+                        break;
+                    case ExpressionType.GreaterThanOrEqual:
+                        comparisonNode = new GteNode();
+                        break;
+                    default:
+                        throw new NotSupportedException($"The binary operator '{be.NodeType}' is not supported");
+                }
+
+                if (be.Left is MemberExpression me)
+                {
+                    if (me.Expression != null && me.Expression.NodeType == ExpressionType.Parameter)
+                    {
+                        var leftName = GetColumnName(me, me.NodeType);
+
+                        comparisonNode.Left.Value = leftName;
+                        comparisonNode.Left.Kind = "property";
+                        comparisonNode.Left.Type = GetKind(me.Type);
+                    }
+                    else
+                    {
+                        throw new NotSupportedException(me + " is null or not a supported type.");
+                    }
+                }
+
+                if (be.Left is ConstantExpression ce)
+                {
+                    comparisonNode.Left.Value = ce.Value;
+                    comparisonNode.Left.Kind = "constant";
+                    comparisonNode.Left.Type = GetKind(ce.Value.GetType());
+                }
+
+                if (be.Right is MemberExpression mo)
+                {
+                    if (mo.Expression != null && mo.Expression.NodeType == ExpressionType.Parameter)
+                    {
+                        var leftName = GetColumnName(mo, mo.NodeType);
+
+                        comparisonNode.Right.Value = leftName;
+                        comparisonNode.Right.Kind = "property";
+                        comparisonNode.Right.Type = GetKind(mo.Type);
+                    }
+                    else
+                    {
+                        throw new NotSupportedException(mo + " is null or not a supported type.");
+                    }
+                }
+
+                if (be.Right is ConstantExpression co)
+                {
+                    comparisonNode.Right.Value = co.Value;
+                    comparisonNode.Right.Kind = "constant";
+                    comparisonNode.Right.Type = GetKind(co.Value.GetType());
+                }
+
+                returnNode = comparisonNode;
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        private static string GetKind(object valueType)
+        {
+            // TODO: Possible with switch statment in our current .NET version?
```

Comment:
It's probably the time to check :)

Reply:
Complains about version should be C# 9.0 or better, so I guess not :)

Reply:
Looks like pattern matching doesn't require a new runtime: https://sergiopedri.medium.com/enabling-and-using-c-9-features-on-older-and-unsupported-runtimes-ce384d8debb

This means we can just bump the C# version and use it.

Reply:
Yep, so let's do it :)